runtime/drivers/clickhouse/embed.go

```diff
@@ -317,18 +350,20 @@ func (e *embedClickHouse) startAndWaitUntilReady(stderr io.Reader) error {
 		case <-timer.C:
 			return fmt.Errorf("clickhouse is not ready: timeout")
 		default:
-			if scanner.Scan() {
-				line := scanner.Text()
-				if strings.Contains(line, "<Error>") {
-					e.logger.Error(line)
-				} else if strings.Contains(line, "Application: Ready for connections") {
-					return nil
-				}
-			} else {
-				if err := scanner.Err(); err != nil {
-					return fmt.Errorf("error reading clickhouse logs: %w", err)
-				}
-				return fmt.Errorf("clickhouse is not ready")
+			if !scanner.Scan() {
```

Comment:
Since this is the `default:` case, doesn't it create a chance for an infinite loop if the scanner is closed? It will just keep calling `scanner.Scan` until the timer runs out.